doc/manual/cookbooks/choosing_nonlinear_solver_scheme/doc.tex

```diff
@@ -0,0 +1,326 @@
+  \draw[thick,->] (0,-pi/2) -- (0,pi/2) node[right]{y};
+    \draw[green, ultra thick,domain=-2.5:2.5] plot (\x, {sin(deg(\x))});
+    \draw[red, thick, domain=-pi:pi] (pi/2,0) -- (pi/2,1);
+    \draw[red,thick,<<->>, domain=-1.4:1] (-2.5,1) -- (2.5,1);
+    \end{tikzpicture}
+}
+\subfigure[][Same as figure a, but now showing the case where the initial guess further away.
+In this case the direction of the next update will be further away from the solution and the
+iteration will diverge.]{
+\begin{tikzpicture}[xscale=1.5,yscale=1.5]
+    \node[draw] at (-1.4,1.4) {find sin(x) = 0};
+    \draw[thick,->] (-2.5,0) -- (2.5,0) node[right]{x};
+  \draw[thick,->] (0,-pi/2) -- (0,pi/2) node[right]{y};
+    \draw[green, ultra thick,domain=-2.5:2.5] plot (\x, {sin(deg(\x))});
+    \draw[red, thick, domain=-pi:pi] (2,0) -- (2,0.909);
+    \draw[red,thick, ->>, domain=2:2.5] plot (\x,-0.41*\x+1.73);
+    \end{tikzpicture}
+}
+\caption{A visualization of the Newton iteration for the nonlinear function {$\sin(x)=0$}.}
+\end{figure}
+
+
+For the Newon iteration, the equation is usually written as 
+\begin{align}
+    \label{sec:cnss:eq3:JdxAxx-b}
+    J \delta x &= A(x) x - b \\
+    &= -F
+\end{align}. In this equation, the right hand side (rhs) ($A(x)x-b$ or $-F$) will be exactly 
+zero when x is the exact solution to the nonlinear problem. Therefore this is also called the 
+residual. $J$ is called the Jacobian, which is a derivative matrix. In our simple example, 
+this would just be the derivative. So we solve for $\delta x$ now instead of $x$:
+
+\begin{align}
+    \delta x_n &= J^{-1}\left(A(x_n) x_n - b\right)
+\end{align}. Next we can compute the new solution with $x_{n+1} = x_n - \delta x_n$. This 
+form of computing an update/correction to the solution instead of computing the full
+solution every time is called a defect correction scheme.
+
+For reason beyond the scope of this cookbook, depending on the rheology model, the Jacobian 
+matrix may sometimes be the equivalent of figure \ref{fig:cnss-fig3-sin0-Newton}, which 
+means that the linear solver will fail. To stabilize this, there are some stabilization 
+options available in ASPECT based on \cite{FBTGS19}, of which the most important is the 
+SPD option. When you need to choose, it is important to know that the unstabilized version
+will always converge at the same speed or faster than the stabilized version. Theoretically 
+this means that you will only want to stabilize when needed. This is an option for the Newton
+solver called the fail-safe. With the fail-safe you can run an unstabilized version of 
+the Newton iteration, and when the linear solver fails, it will automatically turn on all
+the stabilizations for the rest of the timestep and retry the nonlinear iteration. The downside
+is that it may take a long time for the linear solver to fail, especially if it fails every
+timestep. So if you know that the model you are using is causing the linear solver to always
```

Comment:
This sentence is not completely clear to me. Why may it take a long time for the linear solver to fail if if fails every timestep? Do you mean it will take a lot of time when the linear solver fails every timestep?